src/main/java/org/jabref/logic/citationstyle/CSLAdapter.java

```diff
@@ -86,18 +88,26 @@ private void initialize(String newStyle, CitationStyleOutputFormat newFormat) th
     private static class JabRefItemDataProvider implements ItemDataProvider {
 
         private final List<BibEntry> data = new ArrayList<>();
+        private BibDatabase bibDatabase;
 
         /**
          * Converts the {@link BibEntry} into {@link CSLItemData}.
          */
-        private static CSLItemData bibEntryToCSLItemData(BibEntry bibEntry) {
+        private static CSLItemData bibEntryToCSLItemData(BibEntry bibEntry, BibDatabase bibDatabase) {
+            bibDatabase = bibDatabase != null ? bibDatabase : new BibDatabase();
+
             String citeKey = bibEntry.getCitationKey().orElse("");
             BibTeXEntry bibTeXEntry = new BibTeXEntry(new Key(bibEntry.getType().getName()), new Key(citeKey));
 
             // Not every field is already generated into latex free fields
             RemoveNewlinesFormatter removeNewlinesFormatter = new RemoveNewlinesFormatter();
+
+            // adapted from TexBibEntriesResolver
```

Comment:
Please add another comment why the code is necessary. Maybe, it is `// Use information of referenced entry`.

Moreover, I think, the code is wrong for BibLaTeX. It should be externalized to a utility class - or to the BibEntry class? TypedBibEntry?

We should implement the content of Appendix B of the documentation: 

![grafik](https://user-images.githubusercontent.com/1366654/150699762-60a11f95-e539-4ae5-a89f-4233b154941e.png)

Reply:
Ah okay, I see we have already implemented this. It's hidden in the resolvedFieldOrAlias methods.
Problem is that I need to get all Fields (including the empty ones) for the type and therefore I need a bibdatabasecontext. Not sure if I am able to pass it down